sklearn/base.py

```diff
@@ -544,6 +543,36 @@ def fit_predict(self, X, y=None):
         return self.fit(X).predict(X)
 
 
+class OneToOneMixin(object):
+    """Provides get_feature_names for simple transformers
+
+    Assumes there's a 1-to-1 correspondence between input features
+    and output features.
+    """
+
+    def get_feature_names(self, input_features=None):
+        """Get feature names for transformation.
+
+        Returns input_features as this transformation
+        doesn't add or drop features.
+
+        Parameters
+        ----------
+        input_features : array-like of string
+            Input feature names.
+
+        Returns
+        -------
+        feature_names : array-like of string
+            Transformed feature names
+        """
+        if input_features is not None:
+            return input_features
+        else:
+            raise ValueError("Don't know how to get"
```

Comment:
create here dummy features names instead of erroring?

Reply:
if we knew the number of input features we could do that. so right now we can't.